common/database_handler.py

```diff
@@ -71,5 +86,13 @@ def __init__(self):
     def __enter__(self):
         return self.conn
 
-    def __exit__(self, *_):
+    def __exit__(self, exception_type, exception_value, traceback):
         DatabasePoolHandler.return_connection(self.conn)
+        if exception_type is InterfaceError or exception_type is DatabaseError:
```

Comment:
maybe just catch psycopg2.Error instead of both of these? http://initd.org/psycopg/docs/module.html

Reply:
I don't know how to do it, I try a couple of approaches and this one only works, all methods as issubclass() and isinstance() do not work here